controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java

```diff
@@ -438,18 +433,20 @@ private LockedApplication withRotation(LockedApplication application, ZoneId zon
                 application = application.with(rotation.id());
                 store(application); // store assigned rotation even if deployment fails
 
-                GlobalDnsName dnsName = application.get().globalDnsName(controller.system())
```

Comment:
Why can we remove this check?

Reply:
Do you mean the optional check? The endpoints are now expressed as a list instead of an optional, and acting on the empty list doesn't need special handling. It will never be empty here though because a rotation has been assigned in the code above.